library/formdata.inc.php

```diff
@@ -68,14 +110,23 @@ function escape_sort_order($s)
  * the error log. This function should not be used for escaping tables outside the openemr
  * database (should use escape_identifier() function below for that scenario)
  *
- * @param   string        $s       sql column name variable to be escaped/sanitized.
+ * @param   string|array        $s       sql column name(s) variable to be escaped/sanitized.
  * @param   array         $tables  The table(s) that the sql columns is from (in an array).
  * @param   boolean       $long    Use long form (ie. table.colname) vs short form (ie. colname).
  * @return  string                 Escaped table name variable.
  */
 function escape_sql_column_name($s, $tables, $long = false)
 {
+    // If $s is asterisk return asterisk to select all columns
+    if ($s === "*") {
+        return "*";
+    }
 
+    // If $s is an array process with escape_multiple_sql_column_names function
+    if (is_array($s)) {
+        $multiple_columns = escape_multiple_sql_column_names($s, $tables); 
+        return $multiple_columns;
+    }
```

Comment:
I think we can drop the need for the escape_multiple_sql_column_names function by using recursion here:
```
    // If $s is an array process then use recursion to check each column
    if (is_array($s)) {
       $multiple_columns = [];
       foreach ($s as $column) {
           $multiple_columns[] = escape_sql_column_name($column, $tables);
         }
        return implode(", ", $multiple_columns);
    }
```